File: Tip Calculator/main(Java)/java/com/example/tipcalculatorjava/MainActivity.java
package com.example.tipcalculatorjava;

import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SwitchCompat;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {
    EditText edtMoney,edtPercent;
    SwitchCompat toggle;
    TextView txtAns;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        //Id find
        edtMoney=findViewById(R.id.edtMoney);
        edtPercent=findViewById(R.id.edtPercent);
        toggle=findViewById(R.id.toggle);
        txtAns=findViewById(R.id.txtAns);

        //on money value change set the ans to the answer text.
      edtMoney.addTextChangedListener(new TextWatcher() {
          @Override
          public void beforeTextChanged(CharSequence s, int start, int count, int after) {

          }

          @Override
          public void onTextChanged(CharSequence s, int start, int before, int count) {
              String money=s.toString();
              String percent=edtPercent.getText().toString();
              if(!money.isEmpty() && !percent.isEmpty() ){
                  double dMoney=Double.parseDouble(money);
                  double dPercent=Double.parseDouble(percent);
                  txtAns.setText(calculation(dMoney,dPercent,toggle.isChecked()));
              }else{
                  txtAns.setText(getString(R.string.tip_amount));
              }
          }

          @Override
          public void afterTextChanged(Editable s) {

          }
      });

      //If percent value change set ans to the answer text.
      edtPercent.addTextChangedListener(new TextWatcher() {
          @Override
          public void beforeTextChanged(CharSequence s, int start, int count, int after) {

          }

          @Override
          public void onTextChanged(CharSequence s, int start, int before, int count) {
              String percent=s.toString();
              String money=edtMoney.getText().toString();
              if(!money.isEmpty() && !percent.isEmpty() ){
                  double dMoney=Double.parseDouble(money);
                  double dPercent=Double.parseDouble(percent);
                  txtAns.setText(calculation(dMoney,dPercent,toggle.isChecked()));
              }else{
                  txtAns.setText(getString(R.string.tip_amount));
              }

          }

          @Override
          public void afterTextChanged(Editable s) {

          }
      });

      //on switch change set the answer text to rounded ans or not rounded ans
      toggle.setOnClickListener(new View.OnClickListener() {
          @Override
          public void onClick(View v) {
              String money=edtMoney.getText().toString();
              String percent=edtPercent.getText().toString();
              if(!money.isEmpty() && !percent.isEmpty()){
                  double dMoney=Double.parseDouble(money);
                  double dPercent=Double.parseDouble(percent);
                  if(toggle.isChecked()){
                      txtAns.setText(calculation(dMoney,dPercent,true));
                  }else{
                      txtAns.setText(calculation(dMoney,dPercent,false));
                  }
              }
          }
      });

    }

    //Tip Calculation
    public String calculation(double money,double percent,boolean roundUp){
        double ans = (percent / 100) * money;
        if(roundUp) {
           ans=Math.round(ans);
        }
        return getString(R.string.tip_amount)+ans;
    }
}

File: Tip Calculator/main(Kotlin)/java/com/example/tipcalculatorkotlin/MainActivity.kt
package com.example.tipcalculatorkotlin

import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.util.Log
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.SwitchCompat
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        //Id find
        val edtMoney: EditText=findViewById(R.id.edtMoney)
        val edtPercent: EditText=findViewById(R.id.edtPercent)
        val txtAns: TextView =findViewById(R.id.txtAns)
        val toggle: SwitchCompat=findViewById(R.id.toggle)

        //on money value change 1st calculate the ans according the value, then set the ans to the answer text.
        edtMoney.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {

            }

            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {

                val money=s.toString()
                val percent=edtPercent.getText().toString()
                if(money.isNotEmpty() && percent.isNotEmpty()){
                    val dMoney:Double=money.toDoubleOrNull()?:0.0
                    val dPercent:Double=percent.toDouble()
                    txtAns.text = calculation(dMoney,dPercent,toggle.isChecked)
                }else{
                    txtAns.text = getString(R.string.tip_amount)
                }

            }

            override fun afterTextChanged(s: Editable?) {

            }

        })


        //On percent value change 1st calculate the ans according the value, then set ans to the answer text.
        edtPercent.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {

            }

            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
                val percent=s.toString()
                val money=edtMoney.getText().toString()
                if(money.isNotEmpty() && percent.isNotEmpty()){
                    val dMoney=money.toDouble()
                    val dPercent=percent.toDouble()
                    txtAns.text = calculation(dMoney,dPercent,toggle.isChecked)
                }else{
                    txtAns.text = getString(R.string.tip_amount)
                }
            }

            override fun afterTextChanged(s: Editable?) {

            }

        })

        //on switch change 1st get the answer then set the answer text to rounded ans or not rounded ans
        toggle.setOnClickListener {
            val money=edtMoney.getText().toString()
            val percent=edtPercent.getText().toString()
            if(money.isNotEmpty() && percent.isNotEmpty()){
                val dMoney=money.toDouble()
                val dPercent=percent.toDouble()
                if(toggle.isChecked){
                    txtAns.text = calculation(dMoney,dPercent,true)
                }else{
                    txtAns.text = calculation(dMoney,dPercent,false)
                }
            }
        }

    }

    //Tip Calculation
  fun calculation(money:Double,percent:Double,roundUp:Boolean): String{
        var tip = (percent / 100) * money
        if(roundUp) {
            tip=kotlin.math.ceil(tip)
        }
        return getString(R.string.tip_amount)+tip;
    }
}